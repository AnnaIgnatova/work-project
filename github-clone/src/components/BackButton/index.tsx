import { Button } from "antd";
import { useNavigate } from "react-router-dom";
import styles from "./style.module.scss";

export const BackButton = () => {
  const navigate = useNavigate();

  const goBack = () => navigate(-1);

  return (
    <Button onClick={goBack} className={styles.button}>
      Back
    </Button>
  );
};
