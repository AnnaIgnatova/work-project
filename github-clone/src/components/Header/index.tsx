import { useState } from "react";
import { Switch } from "antd";
import styles from "./style.module.scss";
import { Theme } from "../../types/theme";
import { THEME_KEY } from "../../constants/theme";

export const Header: React.FC = () => {
  const [theme, setTheme] = useState<Theme>(
    (localStorage.getItem(THEME_KEY) as Theme) || Theme.DARK
  );

  const onChange = () => {
    const newTheme = theme === Theme.LIGHT ? Theme.DARK : Theme.LIGHT;
    setTheme(newTheme);
    localStorage.setItem(THEME_KEY, newTheme);
  };

  return (
    <header className={`${styles.header} ${theme}`}>
      <h1 className={styles["logo"]}>GitHub Users</h1>
      <Switch onChange={onChange} />
    </header>
  );
};
