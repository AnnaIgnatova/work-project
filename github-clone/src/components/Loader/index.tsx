import loaderPath from "./../../assets/loader.webp";
import styles from './style.module.scss'

export const Loader: React.FC = () => {
  return (
    <div className={styles.loader}>
      <img src={loaderPath} alt="loader" />
    </div>
  );
};
