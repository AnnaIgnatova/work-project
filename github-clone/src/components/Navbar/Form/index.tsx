import {Button, Form, Input} from "antd";
import {useUnit} from "effector-react";
import {
  $createdBy,
  $followersMinCount,
  $userName,
  changeDateCreatedBy,
  changeFollowersMinCount,
  changeUserNameEvent,
} from "../../../store/filters";
import {getUsersEvent} from "../../../store/users";

interface NavbarFormProps {
  isLoading: boolean;
}

export const NavbarForm: React.FC<NavbarFormProps> = ({isLoading}) => {
  const [userName, followers, createdBy, changeName, setFollowers, changeDate] = useUnit([
    $userName,
    $followersMinCount,
    $createdBy,
    changeUserNameEvent,
    changeFollowersMinCount,
    changeDateCreatedBy,
  ]);

  const getUsers = () => getUsersEvent();

  return (
    <Form layout="vertical" disabled={isLoading}>
      <Form.Item label="Name">
        <Input placeholder="Name" aria-label="name-input" value={userName} onChange={changeName} />
      </Form.Item>
      <Form.Item label="Followers count from">
        <Input
          type="number"
          aria-label="followers-input"
          placeholder="Followers count from"
          value={followers}
          onChange={setFollowers}
        />
      </Form.Item>
      <Form.Item label="Created by">
        <Input placeholder="Created by" aria-label="date-input" type="date" value={createdBy} onChange={changeDate} />
      </Form.Item>
      <Form.Item>
        <Button onClick={getUsers} type="primary">
          Search
        </Button>
      </Form.Item>
    </Form>
  );
};
