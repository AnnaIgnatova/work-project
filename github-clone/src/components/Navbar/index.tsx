import { useState } from "react";
import styles from "./style.module.scss";
import { useUnit } from "effector-react";
import { LeftOutlined, RightOutlined } from "@ant-design/icons";
import { Button, Form, Input } from "antd";
import {
  $createdBy,
  $followersMinCount,
  $userName,
  changeDateCreatedBy,
  changeFollowersMinCount,
  changeUserNameEvent,
} from "../../store/filters";
import { getUsersEvent } from "../../store/users";

interface NavbarProps {
  isLoading: boolean;
}

export const Navbar: React.FC<NavbarProps> = ({ isLoading }) => {
  const [showNavbar, setShowNavbar] = useState<boolean>(true);
  const toggleNavbar = () => setShowNavbar(!showNavbar);

  const [userName, followers, createdBy, changeName, setFollowers, changeDate] =
    useUnit([
      $userName,
      $followersMinCount,
      $createdBy,
      changeUserNameEvent,
      changeFollowersMinCount,
      changeDateCreatedBy,
    ]);

  return (
    <div className={styles.navbar}>
      <button
        className={`${styles["navbar__header__button"]}`}
        onClick={toggleNavbar}
      >
        {showNavbar ? (
          <LeftOutlined style={{ fontSize: "24px" }} />
        ) : (
          <RightOutlined style={{ fontSize: "24px" }} />
        )}
      </button>
      <div className={`${!showNavbar ? "hidden" : ""}`}>
        <div className={styles["navbar__header"]}>
          <h3 className={styles["navbar__header__title"]}>Filters</h3>
        </div>
        <Form.Item label="Name">
          <Input
            disabled={isLoading}
            placeholder="Name"
            aria-label="name-input"
            value={userName}
            onChange={changeName}
          />
        </Form.Item>
        <Form.Item label="Followers count from">
          <Input
            disabled={isLoading}
            type="number"
            aria-label="followers-input"
            placeholder="Followers count from"
            value={followers}
            onChange={setFollowers}
          />
        </Form.Item>
        <Form.Item label="Created by">
          <Input
            disabled={isLoading}
            placeholder="Created by"
            aria-label="date-input"
            type="date"
            value={createdBy}
            onChange={changeDate}
          />
        </Form.Item>

        <Button onClick={getUsersEvent} disabled={isLoading}>
          Найти
        </Button>
      </div>
    </div>
  );
};
