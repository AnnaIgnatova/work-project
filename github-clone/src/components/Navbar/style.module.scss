.navbar {
  background: var(--background-color-light);
  max-width: 250px;
  height: calc(100vh - 70px);
  padding: 20px;
  transition: all 0.5s;
  position: relative;
  padding-right: 50px;
}

.navbar__header {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.navbar__header__title {
  font-weight: normal;
}

.navbar__header__button {
  border: none;
  cursor: pointer;
  background-color: transparent;
  position: absolute;
  right: 10px;
  top: 10px;
}

:global(.hidden) {
  display: none;
}

.navbar__container {
  margin-top: 30px;
  display: flex;
  flex-direction: column;
  height: 20vh;
  justify-content: space-between;
}
