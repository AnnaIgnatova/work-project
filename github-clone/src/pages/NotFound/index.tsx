export const NotFoundPage: React.FC = () => {
  return <>not found</>;
};
