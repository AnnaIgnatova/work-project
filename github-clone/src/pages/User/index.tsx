import {useEffect} from "react";
import {useParams} from "react-router-dom";
import {Descriptions, message} from "antd";
import {Loader} from "../../components/Loader";
import {UserData} from "../../store/users/types";
import {BackButton} from "../../components/BackButton";
import {useUnit} from "effector-react";
import {$userData, getUserEvent, getUserFx} from "../../store/user";
import styles from "./style.module.scss";
import Title from "antd/es/typography/Title";

export const UserPage: React.FC = () => {
  const {user} = useParams();
  const [messageApi, contextHolder] = message.useMessage();
  const [userData, getUser, isLoading] = useUnit([$userData, getUserEvent, getUserFx.pending]);

  useEffect(() => {
    getUser({user});
  }, [user, getUser]);

  useEffect(() => {
    if (userData?.message) {
      messageApi.open({
        type: "error",
        content: userData?.message,
      });
    }
  }, [userData]);

  const getDescriptionData = (userData: UserData | null) =>
    userData && !userData?.message
      ? Object.entries(userData).map(([key, value]) => ({
          key,
          label: key,
          children: value,
        }))
      : [];

  return (
    <>
      {contextHolder}
      {isLoading ? (
        <Loader />
      ) : (
        <div className={styles.container}>
          <BackButton />
          {!userData?.message ? (
            <>
              <img src={userData?.avatar_url} alt={userData?.login} className={styles["user__image"]} />
              <Title level={3}>Hello, my name is {userData?.login}. Here some information about me</Title>
              <Descriptions layout="vertical" items={getDescriptionData(userData)} bordered />
            </>
          ) : (
            <>User not found</>
          )}
        </div>
      )}
    </>
  );
};
