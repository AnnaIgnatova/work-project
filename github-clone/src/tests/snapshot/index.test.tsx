import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import {BackButton} from "../../components/BackButton";
import {Navbar} from "../../components/Navbar";
import {NotFoundPage} from "../../pages/NotFound";
import {UserPage} from "../../pages/User";
import { Header } from "antd/es/layout/layout";

describe("snapshot tests", () => {
  it("renders back button correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <BackButton />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders navbar correctly", () => {
    const tree = renderer.create(<Navbar isLoading={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders not found page correctly", () => {
    const tree = renderer.create(<NotFoundPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders user page correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <UserPage />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders header correctly", () => {
    const tree = renderer
      .create(
        <Header />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
