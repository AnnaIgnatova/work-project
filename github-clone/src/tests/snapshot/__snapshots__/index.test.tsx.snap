// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot tests > renders back button correctly 1`] = `
<button
  className="ant-btn css-dev-only-do-not-override-1b0bdye ant-btn-default _button_7912fd"
  disabled={false}
  onClick={[Function]}
  style={{}}
  type="button"
>
  <span>
    Back
  </span>
</button>
`;

exports[`snapshot tests > renders header correctly 1`] = `
<header
  className="ant-layout-header css-dev-only-do-not-override-1b0bdye"
/>
`;

exports[`snapshot tests > renders navbar correctly 1`] = `
<div
  className="_navbar_c18934"
>
  <button
    className="_navbar__header__button_c18934"
    onClick={[Function]}
  >
    <span
      aria-label="left"
      className="anticon anticon-left"
      role="img"
      style={
        {
          "fontSize": "24px",
        }
      }
    >
      <svg
        aria-hidden="true"
        data-icon="left"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
        />
      </svg>
    </span>
  </button>
  <div
    className=""
  >
    <div
      className="_navbar__header_c18934"
    >
      <h3
        className="_navbar__header__title_c18934"
      >
        Filters
      </h3>
    </div>
    <div
      className="ant-form-item css-dev-only-do-not-override-1b0bdye"
    >
      <div
        className="ant-row ant-form-item-row css-dev-only-do-not-override-1b0bdye"
        style={
          {
            "rowGap": undefined,
          }
        }
      >
        <div
          className="ant-col ant-form-item-label css-dev-only-do-not-override-1b0bdye"
          style={{}}
        >
          <label
            className=""
            title="Name"
          >
            Name
          </label>
        </div>
        <div
          className="ant-col ant-form-item-control css-dev-only-do-not-override-1b0bdye"
          style={{}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                aria-label="name-input"
                className="ant-input css-dev-only-do-not-override-1b0bdye ant-input-outlined"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Name"
                style={{}}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-form-item css-dev-only-do-not-override-1b0bdye"
    >
      <div
        className="ant-row ant-form-item-row css-dev-only-do-not-override-1b0bdye"
        style={
          {
            "rowGap": undefined,
          }
        }
      >
        <div
          className="ant-col ant-form-item-label css-dev-only-do-not-override-1b0bdye"
          style={{}}
        >
          <label
            className=""
            title="Followers count from"
          >
            Followers count from
          </label>
        </div>
        <div
          className="ant-col ant-form-item-control css-dev-only-do-not-override-1b0bdye"
          style={{}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                aria-label="followers-input"
                className="ant-input css-dev-only-do-not-override-1b0bdye ant-input-outlined"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Followers count from"
                style={{}}
                type="number"
                value="0"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-form-item css-dev-only-do-not-override-1b0bdye"
    >
      <div
        className="ant-row ant-form-item-row css-dev-only-do-not-override-1b0bdye"
        style={
          {
            "rowGap": undefined,
          }
        }
      >
        <div
          className="ant-col ant-form-item-label css-dev-only-do-not-override-1b0bdye"
          style={{}}
        >
          <label
            className=""
            title="Created by"
          >
            Created by
          </label>
        </div>
        <div
          className="ant-col ant-form-item-control css-dev-only-do-not-override-1b0bdye"
          style={{}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                aria-label="date-input"
                className="ant-input css-dev-only-do-not-override-1b0bdye ant-input-outlined"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Created by"
                style={{}}
                type="date"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="ant-btn css-dev-only-do-not-override-1b0bdye ant-btn-default"
      disabled={false}
      onClick={[Function]}
      style={{}}
      type="button"
    >
      <span>
        Найти
      </span>
    </button>
  </div>
</div>
`;

exports[`snapshot tests > renders not found page correctly 1`] = `"not found"`;

exports[`snapshot tests > renders user page correctly 1`] = `
<div
  className="_loader_5b016b"
>
  <img
    alt="loader"
    src="/src/assets/loader.webp"
  />
</div>
`;
